fe/fe-core/src/main/java/org/apache/doris/planner/OriginalPlanner.java

```diff
@@ -448,6 +448,7 @@ private void checkAndSetTopnOpt(PlanNode node) {
             SortNode sortNode = (SortNode) node;
             PlanNode child = sortNode.getChild(0);
             if (child instanceof OlapScanNode && sortNode.getLimit() > 0
+                    && sortNode.getLimit() <= ConnectContext.get().getSessionVariable().topnOptLimitThreshold
```

Comment:
I'm not sure if session will be null here

Reply:
added check for null